src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypeAlleleCounts.java

```diff
@@ -652,26 +628,17 @@ GenotypeAlleleCounts copy() {
     @SuppressWarnings("unchecked")
     public <T extends Allele> List<T> asAlleleList(final List<T> allelesToUse) {
         Utils.nonNull(allelesToUse, "the input allele list cannot be null");
-        if (allelesToUse.size() < maximumAlleleIndex()) {
-            throw new IllegalArgumentException("the provided alleles to use does not contain an element for the maximum allele ");
-        }
+        Utils.validateArg(allelesToUse.size() >= maximumAlleleIndex(), "the provided alleles to use does not contain an element for the maximum allele");
         if (distinctAlleleCount == 1 ) {
-            if (ploidy == 1) {
-                return Collections.singletonList(allelesToUse.get(sortedAlleleCounts[0]));
-            } else {
-                return Collections.nCopies(ploidy, allelesToUse.get(sortedAlleleCounts[0]));
-            }
+            return Collections.nCopies(ploidy, allelesToUse.get(sortedAlleleCounts[0]));
         } else {
-            final Allele[] myAlleles = new Allele[ploidy];
-            int nextIndex = 0;
+            final List<T> result = new ArrayList<>(ploidy);
             for (int i = 0, ii = 0; i < distinctAlleleCount; i++) {
```

Comment:
You have both `i` and `ii` in this loop -- recommend at least giving `ii` a more descriptive name to avoid confusion.